hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -51,8 +51,26 @@ namespace hardware_interface
  * hardware.
  *
  * The hardware interface map (\ref interfaces_) is a 1-to-1 map between
- * the names of interface types derived from \ref HardwareInterface  and
+ * the names of interface types derived from \ref HardwareInterface and
  * instances of those interface types.
+ * 
+ * A class derived from this base interface represents a robot and stores the 
+ * state of the robot's hardware resources (joints, sensors, actuators) and 
+ * outgoing commands in its data member arrays. The names of these members 
+ * should provide semantic meaning (e.g. pos, vel, eff, cmd). 
+ * Commands from controllers are populated by the controller's update method, 
+ * (controller_interface::ControllerBase::update()).
+ * For each resource a \ref JointStateHandle (for read only joints), 
+ * \ref JointHandle (for read and write joints) or custom handle can be used 
```

Comment:
This question is similar to the [previous one](https://github.com/ros-controls/ros_control/pull/433/files#r405166152). Examples I've seen so far use both (`JointStateHandle` and `JointHandle`) but wouldn't `JointHandle` be enough as it can be used for reading joint state and writing commands?